module/Photo/src/Photo/Service/Photo.php

```diff
@@ -220,7 +221,74 @@ public function movePhoto($photoId, $albumId)
         return true;
 
     }
-
+    /**
+     * Generates the PhotoOfTheWeek and adds it to the list
+     * if at least one photo has been viewed in the specified time. 
+     * The parameters determine the week to check the photos of.  
+     * 
+     * @param \DateTime $begindate
+     * @param \DateTime $enddate
+     * 
+     * @return \Photo\Model\Photo|null
+     */
+    public function generatePhotoOfTheWeek($begindate, $enddate)
+    {
+        $bestPhoto = $this->determinePhotoOfTheWeek($begindate, $enddate);
+        if (is_null($bestPhoto)){
+            return null;
+        }
+        $weeklyPhoto = new WeeklyPhotoModel();
+        $weeklyPhoto->setPhoto($bestPhoto);
+        $weeklyPhoto->setWeek($begindate);
+        $mapper = $this->getWeeklyPhotoMapper();
+        $mapper->persist($weeklyPhoto);
+        $mapper->flush();
+        return $bestPhoto;
+    }
+    
+    /**
+     * Determine which photo is the photo of the week
+     * 
+     * @param \DateTime $begindate
+     * @param \DateTime $enddate
+     * @return \Photo\Model\Photo|null
+     */
+    private function determinePhotoOfTheWeek($begindate, $enddate)
+    {
+        $results = $this->getHitMapper()->getHitsInRange($begindate, $enddate);
```

Comment:
I do not really understand what you mean by making the query bigger, do you mean to test more photos than those that have been visited in the specified date range? Photos that have been photo of the week are excluded in line 265.